### PR TITLE
Adding a test of an empty triggers property removing triggers.

### PR DESCRIPTION
**Pull request by abayer**
Just wanted to make sure we had this and working correctly.

cc @reviewbybees esp @jglick

### CODE REVIEW

**Comment by abayer** [comment]
Need to add resetting of the start/stop list.

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java [hunk 1]
```diff
@@ -296,6 +308,64 @@
         assertEquals("[null, false, null]", MockTrigger.startsAndStops.toString());
     }
 
+    @Test public void scmAndEmptyTriggersProperty() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        // Verify the base case behavior.
+        p.setDefinition(new CpsFlowDefinition("echo 'foo'"));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: :ant: always pass an explicit `sandbox` argument, typically `true` for tests unless you specifically need to test the opposite,

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java [hunk 2]
```diff
@@ -296,6 +308,64 @@
...
+
+        // Make sure the triggers are still empty.
+        assertTrue(p.getTriggers().isEmpty());
+
+        // Now add a trigger.
+        p.setDefinition(new CpsFlowDefinition(
+                (HAVE_SYMBOL ?
+                        "properties([pipelineTriggers([\n"
+                                + "  scm('@daily')])\n" :
+                        // TODO: Make this work with just "spec: '@daily'". Doesn't work for SCMTrigger due to different
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: Well as of Jenkins 2 we do not care any more anyway.

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java [hunk 3]
```diff
@@ -296,6 +308,64 @@
...
+                        // names on DataBoundConstructor...
+                        "properties([pipelineTriggers([[$class: 'SCMTrigger', scmpoll_spec: '@daily']])])\n"
+                ) + "echo 'foo'"));
+
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        // Verify that we're seeing warnings due to running 'properties' in a non-multibranch job.
+        r.assertLogContains(Messages.JobPropertyStep__could_remove_warning(), b);
+        // Verify that we're not seeing warnings for any properties being removed, since there are no pre-existing ones.
+        // Note - not using Messages here because we're not actually removing any properties.
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: :ant: huh? You should still use `Messages` in the test, so that the test will correctly fail if you regress something after editing the message.

**Comment by jglick** [comment]
:bee:

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java (hunk 3 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: I couldn't match the strings via `Messages` when not actually removing anything - i.e., how do I make sure a `Message` with undefined parameters isn't there? Doesn't really make sense.

Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: Oh I see what you are saying—the message takes a format parameter, and we do not have a substring for just the generic part of the message.

Still, you could for example write

```java
String substring = "WARNING: Removing existing job property";
assertThat(Messages.JobPropertyStep__removed_property_warning(""), containsString(substring));
r.assertLogNotContains(substring, b);
```

which ensures that your test is not accidentally weakened.

**Comment by jglick** [comment]
re:bee:

File: src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java (hunk 3 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java]: I like it! Adding that.

**Comment by jglick** [comment]
re:bee:
